engine/crates/engine-v2/schema/src/ids.rs

```diff
@@ -54,5 +55,6 @@ id_newtypes! {
     Schema.resolvers[ResolverId] => Resolver,
     Schema.definitions[DefinitionId] => Definition,
     Schema.input_values[InputValueId] => InputValue,
+    Schema.headers[HeaderId] => SubgraphHeader,
```

Comment:
well, I've put `Subgraph` inside `federation::DataSource` to separate what's specific to a data source and what isn't. Everything in `Schema` is agnostic of the data source, except `data_sources` and `resolvers`. The goal is just to avoid having a gigantic `Schema` struct and avoid name conflicts when data source have similar names for different structs. Like `table` for databases for example. So don't have a strong opinion for `headers` as it'd certainly be re-used for OpenAPI and others. But then it shouldn't be named `SubgraphHeader` if it's shared IMO. 🤷